code/python/settings_file.py

```diff
@@ -157,25 +178,186 @@ def add_settings_file(project_folder, language_code):
+    """
+    gets the book names from the specified bible
+    param lang_code: the language code of the bible
+    param include_file_names: indicates whether to include file names in the list or not
+    param projects: the projects folder
+    return: a list of book names with their corresponding file names
+    """
+    names = []
+    books = (book for book in listdir(project_folder) if ".usfm" in book)
+    for book in books:
+        name = re.sub(BOOK_NUM, "", book)
+        name = re.sub(EXCLUDE_ALPHANUMERICS, "", name)
+        name = re.sub(POST_PART, "", name)
+        names.append((name, book))
+    return names
+
+
+def write_temp_settings_file(project_folder: Path) -> None:
+    """
+    writes a temporary settings file as a place holder to be able to determine the versification
+    param lang_code: the language code of the bible
+    param projects: the projects folder
+    """
+    if not "Settings.xml" in listdir(project_folder):
+        with open(project_folder / "Settings.xml", "w", encoding="utf-8") as set_file:
+            lang_code = str(project_folder).split("\\")[-1]
+            set_file.write(
+                f"""<ScriptureText>
+                <BiblicalTermsListSetting>Major::BiblicalTerms.xml</BiblicalTermsListSetting>
+                <Naming BookNameForm="46-MAT" PostPart="{lang_code}.usfm" PrePart="" />
+                </ScriptureText>"""
+            )
+
+
+def get_corpus(
+    project_folder: Path, vrs_diffs: dict[str, dict[int, dict[int, list[str]]]]
+) -> list[tuple[str, VerseRef, VerseRef]]:
+    """
+    creates a corpus of books found in vrs_diffs
+    param lang_code: the language code of the bible
+    param vrs_diffs: the list of differences in the versifications
+    param projects: the projects folder
+    return: the corpus from the available books in the specified bible
+    """
+    vrs_path = project_folder / "versification"
+    vrs_path.mkdir(parents=True, exist_ok=True)
+    book_names = get_book_names(project_folder)
+    for name in book_names:
+        if name[0] in vrs_diffs.keys():
+            shutil.copyfile(project_folder / name[1], vrs_path / name[1])
+    write_temp_settings_file(project_folder)
+    shutil.copyfile(project_folder / "Settings.xml", vrs_path / "Settings.xml")
+    corpus = ParatextTextCorpus(vrs_path)
+    lines = list(extract_scripture_corpus(corpus, corpus))
+    shutil.rmtree(vrs_path)
+    return lines
+
+
+def get_versification(
+    project_folder: Path,
+    vrs_diffs: dict[str, dict[int, dict[int, list[str]]]],
+) -> tuple[str, list[str]]:
+    """
+    gets the versification of the given bible
+    param lang_code: the language code of the bible
+    param vrs_diffs: the list of differences in the versifications
+    param log: the list of information about downloads
+    param projects: the projects folder
+    return: the versification of the given bible
+    """
+    lines = get_corpus(project_folder, vrs_diffs)
+    versifications = [
```

Comment:
You could use `versification_to_num.keys()`.

Reply:
For most of the sources I did not notice that they had any deuterocanonical/apocryphal content. I included the script I used to write vrs_diffs.yaml so it should not be too difficult to add those differences if that would be beneficial. I will change it to versification_to_num.keys(), thanks for pointing that out